File: dsd/frameworks_mvc_full/aspnet/Controllers/ProdutoController.cs
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;

using App.Data;
using App.Models;

namespace App.Controllers;

public class ProdutoController : Controller
{
    private readonly AppDbContext _db;

    public ProdutoController(AppDbContext db)
    {
        _db = db;
    }

    public IActionResult Index()
    {
        ViewBag.Categorias =
            _db.Categorias.ToList();

        var produtos = _db.Produtos
            .Include(p => p.Categoria)
            .ToList();

        return View(produtos);
    }

    [HttpPost]
    public IActionResult Salvar(
        string nome,
        int categoriaId)
    {
    	var produto = new Produto
        {
            Nome = nome,
            CategoriaId = categoriaId
        };

        _db.Produtos.Add(produto);
        _db.SaveChanges();

        return RedirectToAction("Index");
    }
}

File: dsd/frameworks_mvc_full/laravel/app/Http/Controllers/ProdutoController.php
<?php

namespace App\Http\Controllers;

use Illuminate\Http\Request;
use App\Models\Produto;
use App\Models\Categoria;

class ProdutoController
{
    public function index()
    {
        $produtos = Produto::with(
            'categoria'
        )->get();

        $categorias = Categoria::all();

        return view(
            'produtos',
            compact(
                'produtos',
                'categorias'
            )
        );
    }

    public function salvar(
        Request $request
    )
    {
        Produto::create([

            'nome' =>
                $request->nome,

            'categoria_id' =>
                $request->categoria_id

        ]);

        return redirect('/');
    }
}